### PR TITLE
Add Recordio File Reader

### PR DESCRIPTION
**Pull request by kexinzhao**

### CODE REVIEW

File: src/tape.cc [hunk 1]
```diff
@@ -125,7 +125,7 @@ void Tape::AddOp(const std::string &type,
                  const VariableHandleMap &in_vars,
                  VariableHandleMap out_vars,
                  const framework::AttributeMap &attrs) {
-  InferShapeAndVarType(type, in_vars, &out_vars, attrs);
+  // InferShapeAndVarType(type, in_vars, &out_vars, attrs);
```
Comment (tonyyang-svail) [inline src/tape.cc]: Please uncomment this

File: src/test_tape.cc [hunk 2]
```diff
@@ -24,6 +25,37 @@ using paddle::tape::Fill;
 using paddle::tape::reset_global_tape;
 using paddle::tape::get_global_tape;
 
+VariableHandle create_recordio_file_reader(std::string filename,
```
Comment (tonyyang-svail) [inline src/test_tape.cc]: move this function to `function.h`

File: src/tape.cc (hunk 1 above)
Comment (kexinzhao) [reply src/tape.cc]: Done

File: src/test_tape.cc (hunk 2 above)
Comment (kexinzhao) [reply src/test_tape.cc]: Done

**Review by tonyyang-svail — APPROVED** [review]
LGTM

**Pull request merged**